graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/SuggestionsResource.java

```diff
@@ -101,14 +112,19 @@ private Set<String> adaptStreams(Set<String> streams, SearchUser searchUser) {
     }
 
     private RelativeRange defaultTimeRange() {
-        try {
-            return RelativeRange.create(300);
-        } catch (InvalidRangeParametersException e) {
-            throw new RuntimeException(e);
-        }
+        return RelativeRange.create(300);
     }
 
     private ImmutableSet<String> loadAllAllowedStreamsForUser(SearchUser searchUser) {
         return permittedStreams.load(searchUser);
     }
+
+    private SuggestionFieldType getFieldType(Set<String> streams, TimeRange timerange, final String fieldName) {
+        final Set<MappedFieldTypeDTO> fieldTypes = mappedFieldTypesService.fieldTypesByStreamIds(streams, timerange);
+        return fieldTypes.stream().filter(f -> f.name().equals(fieldName))
+                .map(field -> field.type().type())
+                .findFirst()
```

Comment:
I am not sure if "findFirst" is a good solution here.
If a field has different types in different indices, mainly if we mix textual and other types, I'd say we should go for OTHER or even new (MIXED?) enum value to trigger scripted variant of the query.

Reply:
You are right.
Maybe MappedFieldTypesServiceImpl should return Map<String, MappedFieldTypeDTO> instead of Set<MappedFieldTypeDTO>, to precisely announce that it prepares only one DTO per field name... But that's a question for different task...

Reply:
I think the MappedFieldTypesServiceImpl and its types would deserve some refactoring. And agree that this is a goal for another task.